osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -146,17 +185,11 @@ protected override void LoadComplete()
 
             protected float GetSizeFor(int combo)
             {
-                float size = defaultFlashlightSize * sizeMultiplier;
+                if (!comboBasedSize) return appliedFlashlightSize;
 
-                if (comboBasedSize)
-                {
-                    if (combo >= 200)
-                        size *= 0.8f;
-                    else if (combo >= 100)
-                        size *= 0.9f;
-                }
+                float changeSizeComboReachedTimesLimited = MathF.Min(maxChangeSizeTimes, MathF.Floor(combo / changeSizeCombo));
 
-                return size;
+                return appliedFlashlightSize * (1 - changeSizeComboReachedTimesLimited * changeSizeDecreaseRatio);
```

Comment:
to match above suggestion
```suggestion
                return appliedFlashlightSize * (1 - sizeChangeCount * changeSizeDecreaseRatio);
```

Reply:
done.